text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,125 @@
+    - The issue will be opened against the `https://github.com/buildpacks/registry` repo, which is *not* the index. The `buildpacks/registry` defines the [namespace ownership as described in RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md#namespace-ownership).
+    - The issue body will contain structured data that defines the buildpack id, version, digest, and url.
+1. A Github action will detect the new issue and do the following:
+    - If this is the first time the `namespace` is used, it will add the namespace to the `buildpacks/registry` repo with the Github user who opened the issue as the owner.
+    - If this is *not* the first time the `namespace` is used, it will confirm that the Github user who opened the issue is an owner of the buildpack.
+    - Create and merge a PR against the `https://github.com/buildpacks/registry-index` repo using a Gitub token (i.e. all commits in that repo will be made by the same "user"). The PR will add the buildpack version described in the issue.
+    - Close the Github issue.
+
+Options:
+
+* `--headless` - this option will use the Github API to create the Github Issue. However, it will require a Github API token from the user.
+
+## `pack yank-buildpack <buildpack-id-and-version>`
+
+Arguments:
+
+* `buildpack-id-and-version` - a buildpack ID and version in the form `<id>@<version>`
+
+Steps:
+
+1. Opens a link to a new Github Issue in the user's browser
+    - The user must be logged into Github
+    - The link will use [Github query parameters](https://help.github.com/en/github/managing-your-work-on-github/about-automation-for-issues-and-pull-requests-with-query-parameters) to pre-populate the title, body, and labels.
+    - The issue will be opened against the `https://github.com/buildpacks/registry` repo.
+    - The issue body will contain structured data that defines the buildpack id and version to yank.
+1. A Github action will detect the new issue and do the following:
+    - Confirm that the Github user who opened the issue is an owner of the buildpack.
+    - Create and merge a PR against the `https://github.com/buildpacks/registry-index` repo using a Gitub token (i.e. all commits in that repo will be made by the same "user"). The PR will set `yanked=true` for the buildpack version described in the issue.
+    - Close the Github issue.
+
+Options:
+
+* `--undo` - this option will execute the same flow, but set `yanked=false`
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* Github is a single-point-of-failure
+* The user must have a Github account
+* The user must publish via a browser or provide a Github API token to the CLI
+* "publish" is ambigious (we use it in `pack package-buildpack --publish` to mean pushing an OCI image to a Docker registry).
```

Comment:
i thought about this too, but in some sense `--push` for those commands isn't accurate, since it's not actually pushing a local image (since it's writing directly to the registry). I also worry about the breaking change.

Reply:
release?